packages/intrepid2/unit-test/Shared/ArrayTools/test_01.hpp

```diff
@@ -91,11 +91,11 @@ namespace Intrepid2 {
       Teuchos::oblackholestream oldFormatState;
       oldFormatState.copyfmt(std::cout);
 
-      typedef typename
-        Kokkos::Impl::is_space<DeviceSpaceType>::host_mirror_space::execution_space HostSpaceType ;
+      using DeviceExecSpaceType = typename DeviceType::execution_space;
+      using HostExecSpaceType = Kokkos::DefaultHostExecutionSpace;
```

Comment:
Oh, now I understand what you were saying (I hadn't seen your latest).  Yes, this is fine now.